scan-engine/src/main/java/de/tillhub/scanengine/zebra/ZebraBarcodeScanner.kt

```diff
@@ -89,17 +97,52 @@ internal class ZebraBarcodeScanner(
         context.startActivity(intent)
     }
 
-    override fun dcssdkEventCommunicationSessionEstablished(scanner: DCSScannerInfo?) {
-        if (scanner == null) return
-        sdkHandler.dcssdkEnableAutomaticSessionReestablishment(true, scanner.scannerID)
+    override fun observeScanners(): Flow<List<Scanner>> = availableScannersFlow
+
+    override suspend fun connect(scannerId: String): ScannerResponse {
+        return try {
+            val result = sdkHandler.dcssdkEstablishCommunicationSession(scannerId.toInt())
+            when (result) {
+                DCSSDKDefs.DCSSDK_RESULT.DCSSDK_RESULT_SUCCESS -> ScannerResponse.Success
+                DCSSDKDefs.DCSSDK_RESULT.DCSSDK_RESULT_FAILURE ->
+                    ScannerResponse.Error.Connect(R.drawable.classic_discoverable)
+
+                else -> ScannerResponse.Error.NotFound
+            }
+        } catch (e: NumberFormatException) {
+            Timber.e(e, "Error connecting to scanner with ID: $scannerId")
```

Comment:
Do we need Timber?

Reply:
Not necessary for connect but kept it for disconnect as we don't have return type